src/request.ts

```diff
@@ -411,7 +426,7 @@ export class OData {
         url += `(${id})`
         break;
       case "string":
-        url += `('${id}')`
+        url += `(${id})`
```

Comment:
Please do NOT change this logic, if for `S/4HANA` have different logic from standard odata, add new option to `BatchRequestOptions` and `if-else` to set `url` value

Reply:
Compound keys are structured as `OrderItems(OrderId=1234,ProductId=1234)`. Grammar is defined [here](http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/abnf/odata-abnf-construction-rules.txt). 

Example of usage can be found [here](http://docs.oasis-open.org/odata/odata/v4.0/errata03/os/complete/part2-url-conventions/odata-v4.0-errata03-os-part2-url-conventions-complete.html#_Toc453752343).

I'll remove it from this PR to create a new one only for it.

Reply:
```suggestion
        url += `('${id}')`
```